src/api/tasks/stage.ts

```diff
@@ -0,0 +1,213 @@
+      const finalTasks = tasks.data?.filter((task) => task.Role === "Final");
+      const orgTasks = tasks.data?.filter((task) => task.Role === "Org");
+      const pcoTasks = tasks.data?.filter((task) => task.Role === "PCO");
+      const distributionTasks = tasks.data?.filter(
+        (task) => task.Role === "Distributor"
+      );
+
+      switch (stage) {
+        case "Peer Review":
+          if (prTasks) {
+            const approvedTasks = prTasks.filter(
+              (task) => task.Status === "Approved"
+            );
+            if (prTasks.length === approvedTasks.length) {
+              updateNeeded = "Final Review";
+            }
+          }
+          break;
+
+        case "Final Review":
+          if (finalTasks) {
+            const approvedTasks = finalTasks.filter(
+              (task) => task.Status === "Approved"
+            );
+            if (finalTasks.length === approvedTasks.length) {
+              if (orgTasks?.length) {
+                updateNeeded = "Organizational Review";
+              } else {
+                updateNeeded = "Approval";
+              }
+            }
+          }
+          break;
+
+        case "Organizational Review":
+          if (orgTasks) {
+            const approvedTasks = orgTasks.filter(
+              (task) => task.Status === "Approved"
+            );
+            if (orgTasks.length === approvedTasks.length) {
+              updateNeeded = "Approval";
+            }
+          }
+          break;
+
+        case "Approval":
+          if (pcoTasks) {
+            const approvedTasks = pcoTasks.filter(
+              (task) => task.Status === "Approved"
+            );
+            if (pcoTasks.length === approvedTasks.length) {
+              updateNeeded = "Distribution";
+            }
+          }
+          break;
+
+        case "Distribution":
+          if (distributionTasks) {
+            const approvedTasks = distributionTasks.filter(
+              (task) => task.Status === "Approved"
+            );
+            if (distributionTasks.length === approvedTasks.length) {
+              updateNeeded = "Distributed";
+            }
+          }
+          break;
+      }
+
+      if (!updateNeeded) {
+        return Promise.resolve();
+      }
+      sendTaskEmails.mutate(updateNeeded);
+      return subWebContext(String(subSite))
+        .web.lists.getByTitle("pcols")
+        .items.getById(pcolId)
+        .update({ Stage: updateNeeded });
+    },
+    onSuccess: () =>
+      queryClient.invalidateQueries({
+        queryKey: ["PCOL", subSite, pcolId],
+      }),
+  });
+};
+
+const useStageUpdateEmail = (subSite: string, pcolId: number) => {
+  const sendEmail = useSendEmail();
+  const pcol = usePCOL(subSite, pcolId);
+  const tasks = useTasks(subSite, pcolId);
+
+  return useMutation({
+    mutationFn: async (stage: string) => {
+      const parallelTasks = tasks.data?.filter(
+        (task) => task.Role === "Parallel" || task.Role === "Serial"
```

Comment:
Should just be parallel tasks
```suggestion
        (task) => task.Role === "Parallel"
```